src/options/theme.rs

```diff
@@ -86,24 +86,23 @@ fn is_no_syntax_highlighting_syntax_theme_name(theme_name: &str) -> bool {
 fn get_is_light_mode_and_syntax_theme_name(
     theme_arg: Option<&String>,
     bat_theme_env_var: Option<&String>,
-    light_mode: bool,
+    light_mode: Option<bool>,
 ) -> (bool, String) {
-    match (theme_arg, bat_theme_env_var, light_mode) {
-        (None, None, false) => (false, DEFAULT_DARK_SYNTAX_THEME.to_string()),
-        (Some(theme_name), _, false) => (is_light_syntax_theme(theme_name), theme_name.to_string()),
-        (None, Some(theme_name), false) => {
-            (is_light_syntax_theme(theme_name), theme_name.to_string())
-        }
-        (None, None, true) => (true, DEFAULT_LIGHT_SYNTAX_THEME.to_string()),
-        (Some(theme_name), _, is_light_mode) => (is_light_mode, theme_name.to_string()),
-        (None, Some(theme_name), is_light_mode) => (is_light_mode, theme_name.to_string()),
+    let theme_arg = theme_arg.or(bat_theme_env_var);
+    match (theme_arg, light_mode) {
+        (Some(theme_name), None) => (is_light_syntax_theme(theme_name), theme_name.to_string()),
+        (Some(theme_name), Some(is_light_mode)) => (is_light_mode, theme_name.to_string()),
+        (None, None | Some(false)) => (false, DEFAULT_DARK_SYNTAX_THEME.to_string()),
+        (None, Some(true)) => (true, DEFAULT_LIGHT_SYNTAX_THEME.to_string()),
     }
 }
 
-fn get_is_light(opt: &cli::Opt) -> bool {
-    get_is_light_opt(opt)
-        .or_else(|| should_detect_dark_light(opt).then(detect_light_mode))
-        .unwrap_or_default()
+fn get_is_light(opt: &cli::Opt) -> Option<bool> {
+    get_is_light_opt(opt).or_else(|| {
+        should_detect_dark_light(opt)
+            .then(detect_light_mode)
+            .flatten()
+    })
 }
```

Comment:
In my opinion true/false is clear, but a tri-state should get a name, in this case it doesn't even need `PartialEq`. Also `get_is_light_opt()` can be moved into the function.

Reply:
What would you think about an `enum ColorMode { Dark, Light }`? This could be used in place of the bool everywhere `is_light` appears. I think `Option<ColorMode>` is then pretty clear as well.

Reply:
I pushed [b33459b](https://github.com/dandavison/delta/pull/1843/commits/b33459b70399d49ce1cddfb145dac5a803cffb84) that implements an enum and replaces the "is light mode" bool everywhere. It's easy to revert if you disagree with the change :)

Reply:
The enum feels like an improvement to me. I think it is a bool mainly because I was very new to Rust when I added it.